pallets/asset-manager/src/lib.rs

```diff
@@ -351,7 +475,130 @@ pub mod pallet {
 			asset_id: T::AssetId,
 			num_assets_weight_hint: u32,
 		) -> DispatchResult {
-			T::AssetModifierOrigin::ensure_origin(origin)?;
+			T::ForeignAssetModifierOrigin::ensure_origin(origin)?;
+
+			// Grab supported assets
+			let mut supported_assets = SupportedFeePaymentAssets::<T>::get();
+
+			ensure!(
+				num_assets_weight_hint >= (supported_assets.len() as u32),
+				Error::<T>::TooLowNumAssetsWeightHint
+			);
+
+			let asset_type =
+				AssetIdType::<T>::get(&asset_id).ok_or(Error::<T>::AssetDoesNotExist)?;
+
+			// Remove from AssetIdType
+			AssetIdType::<T>::remove(&asset_id);
+			// Remove from AssetTypeId
+			AssetTypeId::<T>::remove(&asset_type);
+			// Remove previous asset type units per second
+			AssetTypeUnitsPerSecond::<T>::remove(&asset_type);
+
+			// Only if the old asset is supported we need to remove it
+			if let Ok(index) = supported_assets.binary_search(&asset_type) {
+				supported_assets.remove(index);
+			}
```

Comment:
only updated in this branch so only need to write it in this branch
```suggestion
			if let Ok(index) = supported_assets.binary_search(&asset_type) {
				supported_assets.remove(index);
				<SupportedFeePaymentAssets::<T>::put(supported_assets);
			}
```
and would remove the put a few lines below